### PR TITLE
Upgrades BYOND to 511

### PR DESCRIPTION
**Pull request by Cyberboss**
See #26056 

:cl:
experiment: /tg/station now requires BYOND 511 or greater
/:cl:

### CODE REVIEW

File: config/config.txt [hunk 1]
```diff
@@ -307,7 +307,7 @@ ALLOW_MAP_VOTING 1
 ## These messages will be followed by one stating the clients current version and the required version for clarity.
 #CLIENT_WARN_VERSION 511
 #CLIENT_WARN_MESSAGE Byond is really close to releasing 511 beta as the stable release, please take this time to try it out. You can set the framerate your client runs at, which makes the game feel very different and cool. Shortly after its release we will end up using 511 client features and you will be forced to update.
-CLIENT_ERROR_VERSION 510
+CLIENT_ERROR_VERSION 511
 CLIENT_ERROR_MESSAGE Your version of byond is not supported. Please upgrade.
```
Comment (MrStonedOne) [inline config/config.txt]: If you make the client use a beta version, you MUST specify this in the error message as they will go download the stable version and get confused.

File: code/_compile_options.dm [hunk 2]
```diff
@@ -63,11 +63,11 @@
 #endif
 
 //Update this whenever you need to take advantage of more recent byond features
-#define MIN_COMPILER_VERSION 510
+#define MIN_COMPILER_VERSION 511
 #if DM_VERSION < MIN_COMPILER_VERSION
 //Don't forget to update this part
 #error Your version of BYOND is too out-of-date to compile this project. Go to byond.com/download and update.
-#error You need version 510 or higher
+#error You need version 511 or higher
```
Comment (MrStonedOne) [inline code/_compile_options.dm]: If you make the server use a beta version, you MUST specify this in the error message as they will go download the stable version and get confused.

**Comment by Iamgoofball** [comment]
# this deploys byond telemetry to detect what you do in byond

:-1: my privacy is void

File: .travis.yml [hunk 3]
```diff
@@ -3,8 +3,8 @@ sudo: false
 
 env:
     global:
-    - BYOND_MAJOR="510"
-    - BYOND_MINOR="1346"
+    - BYOND_MAJOR="511"
+    - BYOND_MINOR="1379"
```
Comment (MrStonedOne) [inline .travis.yml]: change this to 1380, the version that will have the fix. its now out now, so travis will fail, but this shouldn't get merged until that goes live.

**Comment by Iamgoofball** [comment]
what the fuck MSO you're HELPING SUPPORT (((BYOND)))'S ABUSE OF ITS PLAYERS

**Comment by Incoming5643** [comment]
Forcing people to use the beta branch of byond to even access our servers is a definitive no. Keep waiting for that stable release.

**Comment by lzimann** [comment]
We already force people to use beta if they want to actually play the game though.

**Comment by optimumtact** [comment]
@Incoming5643 the server has been 511 required for quite some time

**Comment by Incoming5643** [comment]
Unless something's changed recently you can still play the game in 510, you just can't use client fps. Planes are in 510.

@optimumtact there's a difference between asking a small group of people (server owners) to do something and asking everyone to do it. Even if doing so is easy, it is not the default for anyone. This means that most players are not on the beta branch and some subset of those people wouldn't figure out how to get on the beta branch.

Also putting players on the beta branch means that down the line they could update to an unstable beta build then fill the issue tracker with transient byond bugs that no one can reproduce. Not a great situation.

**Comment by optimumtact** [comment]
this pr is pointless by the way, just wait for 511 to go stable

**Comment by optimumtact** [comment]
Like I said, wait for 511 to go stable , we don't need the pixel scaling that badly

**Pull request closed without merging**